tools/trace-plotter/main.go

```diff
@@ -54,21 +58,93 @@ func main() {
 	var traces []Trace
 	l.GetTraces(*pageSize, &traces)
 
-	page := components.NewPage()
-	page.PageTitle = "Trace Plots"
-
 	parsedE2ETraces, e2eDurations := ParseTraces(traces, "e2e")
-	parsedSystemTraces, systemDurations := ParseTraces(traces, "system")
-	page.AddCharts(
-		PlotGraph(parsedE2ETraces, e2eDurations, *zipkinURL, "e2e"),
-		PlotGraph(parsedSystemTraces, systemDurations, *zipkinURL, "system"),
-	)
-
-	if filepath.Ext(*fileName) != ".html" {
-		*fileName += ".html"
+	if !*skipPlotting {
+		page := components.NewPage()
+		page.PageTitle = "Trace Plots"
+
+		parsedSystemTraces, systemDurations := ParseTraces(traces, "system")
+
+		page.AddCharts(
+			PlotGraph(parsedE2ETraces, e2eDurations, *zipkinURL, "e2e"),
+			PlotGraph(parsedSystemTraces, systemDurations, *zipkinURL, "system"),
+		)
+
+		if filepath.Ext(*fileName) != ".html" {
+			*fileName += ".html"
+		}
+		f, _ := os.Create(*fileName)
+		if err := page.Render(io.MultiWriter(f)); err != nil {
+			log.Errorf("Error rendering plot: %s", err)
+		}
+	}
+
+	if *download != "" {
+		downloadTraces(parsedE2ETraces)
+	}
```

Comment:
Coding style suggestion

```suggestion
        if *download != "" {
		downloadTraces(parsedE2ETraces)
	}
	if *skipPlotting {
	    return
	}
	page := components.NewPage()
	page.PageTitle = "Trace Plots"

	parsedSystemTraces, systemDurations := ParseTraces(traces, "system")

	page.AddCharts(
		PlotGraph(parsedE2ETraces, e2eDurations, *zipkinURL, "e2e"),
		PlotGraph(parsedSystemTraces, systemDurations, *zipkinURL, "system"),
	)

	if filepath.Ext(*fileName) != ".html" {
		*fileName += ".html"
	}
	f, _ := os.Create(*fileName)
	if err := page.Render(io.MultiWriter(f)); err != nil {
		log.Errorf("Error rendering plot: %s", err)
	}
```